### PR TITLE
Fixed the regression bug for aggregates with no window.

### PR DESCRIPTION
**Pull request by hjafarpour**

### CODE REVIEW

**Comment by apurvam** [comment]
What bug did you fix? When was the regression introduced? Also, can we add unit tests? I see only integration tests for the fix.

**Comment by apurvam** [comment]
Also worth including in the description: how does the bug manifest? What are the symptoms? Do tests fail? Do queries give wrong results sometimes?

**Comment by hjafarpour** [comment]
The but was introduced when we switched to the new Streams API in 1.0. The state store serde was not set correctly. This resulted in query failure because of incorrect serde settings.
The tests did not catch it since we did not have a test with GROUP BY without window. I added an integration test to capture this case too.

File: ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKGroupedStream.java [hunk 1]
```diff
@@ -61,13 +61,13 @@ public SchemaKTable aggregate(final Initializer initializer,
     final KTable aggKtable;
     if (windowExpression != null) {
       final Materialized<String, GenericRow, ?> materialized
-          = Materialized.<String, GenericRow, WindowStore<Bytes, byte[]>>as(storeName)
+          = Materialized.<String, GenericRow, WindowStore<String, GenericRow>>as(storeName)
```
Comment (dguy) [inline ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKGroupedStream.java]: This is actually incorrect. It should be `WindowStore<Bytes, byte[]>` the underlying store is always of type `<Bytes, byte[]>` now.

It should be:
```
Materialized.<String, GenericRow, WindowStore<Bytes, byte[]>>as(storeName)
   .withKeySerde(Serdes.String())
   .withValueSerde(topicValueSerDe)

File: ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java [hunk 2]
```diff
@@ -69,6 +69,52 @@ public void after() throws Exception {
...
+      final GenericRow actual = aggregateResults.get("ITEM_1");
+      return expected.equals(actual);
+    }, 60000, "didn't receive correct results within timeout");
+
+    AdminClient adminClient = AdminClient.create(testHarness.ksqlConfig.getKsqlStreamConfigProps());
+    KafkaTopicClient topicClient = new KafkaTopicClientImpl(adminClient);
+
+    Set<String> topicBeforeCleanup = topicClient.listTopicNames();
+
+    assertThat("Expected to have 5 topics instead have : " + topicBeforeCleanup.size(),
```
Comment (dguy) [inline ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java]: `assertThat("..", topicBeforeCleanup.size(), equalTo(5))`

File: ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java [hunk 3]
```diff
@@ -69,6 +69,52 @@ public void after() throws Exception {
...
+               topicBeforeCleanup.size() == 5);
+    QueryMetadata queryMetadata = ksqlContext.getRunningQueries().iterator().next();
+
+    queryMetadata.close();
+    Set<String> topicsAfterCleanUp = topicClient.listTopicNames();
+
+    assertThat("Expected to see 3 topics after clean up but seeing " + topicsAfterCleanUp.size
```
Comment (dguy) [inline ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java]: `assertThat("..", topicsAfterCleanup.size(), equalTo(3))`

File: ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKGroupedStream.java (hunk 1 above)
Comment (hjafarpour) [reply ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKGroupedStream.java]: @dguy thanks for the correction. Fixed it.

File: ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java (hunk 2 above)
Comment (hjafarpour) [reply ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java]: Done.

File: ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java (hunk 3 above)
Comment (hjafarpour) [reply ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java]: Done.

**Pull request merged**